examples/ArduinoWiFi101/ArduinoWiFi101.ino

```diff
@@ -62,10 +66,6 @@ void loop() {
   }
 }
 
-void messageReceived(String topic, String payload, char * bytes, unsigned int length) {
-  Serial.print("incoming: ");
-  Serial.print(topic);
-  Serial.print(" - ");
-  Serial.print(payload);
-  Serial.println();
+void messageReceived(String topic, String payload) {
```

Comment:
From my experience most users do not use binary data (that includes the null terminator in between) but just use strings. The library null terminates all payload ins the background to prevent overflows. If the user needs to deal with binary data he can register the `onAdvancadeMessage()` that includes the raw buffer and its length as the third and fourth argument. Do you think that is too complicated?

A `Buffer` class would be amazing!

@tigoe That is amazing, I'm honored! Please comment as much as you like. I think the `const String&` was more about the Strings that get passed into the library.